sphinx/writers/latex.py

```diff
@@ -1342,6 +1342,9 @@ def depart_thead(self, node):
     def visit_tbody(self, node):
         # type: (nodes.Node) -> None
         self.pushbody(self.table.body)  # Redirect body output until table is finished.
+        if self.footnote_restricted:
+            # releases footnotetexts from header in first non header cell
+            self.unrestrict_footnote(node.parent.parent)
```

Comment:
You're right. The table nodes are always structured as `table > tgroup > tbody`.
But I feel it is unbalance to start restriction on `visit_table` and to unrestrict on `visit_tbody`.
Basically, a symmetry is important to make readability.

To my understanding, the restriction is needed during processing a table caption.
So it would be nice like following:
```
def visit_title(self, node):
    ...
        elif isinstance(parent, nodes.table):  # table caption!
            # Redirect body output until title is finished.
            self.pushbody([])
            self.restrict_footnote(node)  # start footnote-restriction for table caption

def depart_title(self, node):
    ...
    if isinstance(node.parent, nodes.table):
        self.table.caption = self.popbody()  # store a caption text of the table

        self.pushbody([])   # add buffer for \footnotetext's temporarily
        self.unrestrict_footnote(node)
        self.table.caption_footnotes = self.popbody()  # store footnotes inside the caption

def depart_table(self, node):
    ...
    # output footnotes mark-ups on self.table.caption_footnote here
    # (using self.body.append() or templates)
    ...
```

What do you think?